examples/Main.hs

```diff
@@ -48,6 +56,53 @@ example2 = plam $ \x -> pmatch x $ \case
   PLeft n -> n + 1
   PRight n -> n - 1
 
+data SampleRecord f = SampleRecord
+  { sampleBool :: f PBool
+  , sampleInt :: f PInteger
+  , sampleString :: f PString
+  }
+
+--type instance ScottEncoded SampleRecord a = PBool :--> PInteger :--> PString :--> a
+
+sampleRecord :: PRecord SampleRecord s
+sampleRecord =
+  PRecord
+    SampleRecord
+      { sampleBool = pcon PFalse
+      , sampleInt = 6
+      , sampleString = "Salut, Monde!"
+      }
+
+sampleRecur :: Term s (PRecord SampleRecord)
+sampleRecur =
+  letrec $
+    const
+      SampleRecord
+        { sampleBool = pcon PTrue
+        , sampleInt = 12
+        , sampleString = "Hello, World!"
+        }
+
+data EvenOdd f = EvenOdd
+  { even :: f (PInteger :--> PBool)
+  , odd :: f (PInteger :--> PBool)
+  }
+
+type instance ScottEncoded EvenOdd a = (PInteger :--> PBool) :--> (PInteger :--> PBool) :--> a
+
+evenOddBuilder :: EvenOdd (Term s) -> EvenOdd (Term s)
+evenOddBuilder EvenOdd {even, odd} =
+  EvenOdd
+    { even = plam $ \n -> pif (n #== 0) (pcon PTrue) (odd #$ n - 1)
+    , odd = plam $ \n -> pif (n #== 0) (pcon PFalse) (even #$ n - 1)
+    }
+
+evenOdd :: Term s (PRecord EvenOdd)
+evenOdd = letrec evenOddBuilder
```

Comment:
This isn't really an issue, but I am wondering why you didn't just inline `evenOddBuilder` into this function. Wouldn't that be cleaner?

Reply:
I just wanted to have its type signature in the clear, that's the only reason. I demoted the function with ccbfd32.